### DISCUSSION TITLE
How to solve UR10e urdf visualisation blown up?

### DISCUSSION THREAD

**Discussion by Charlie0257** [haosulab/SAPIEN#137] [Q&A] [answered]
When I load the urdf of shadowhand and UR10e model in Sapien, the visualization of UR10e blows up but  shadowhand doesn't, and warns of the following error——
`[SAPIEN] [error] Cylinder collision is not supported. Replaced with a capsule`

Could you give me some suggestions or update the UR10e urdf into the project?
Thanks for any help :) @fbxiang @daerduoCarey

**Comment by fbxiang** [comment]
The cylinder geometry is not currently supported in SAPIEN, but it will be in the next version. Currently the workaround is to replace any cylinder with a corresponding mesh geometry.

**Comment by Charlie0257** [comment]
Thanks for your reply:) @fbxiang 

Could you provide detailed instructions on how to handle cylinders in mesh files? 
I can't find the word cylinder in the .dae file :(

**Reply by fbxiang** [reply]
The cylinder is specified in the URDF file. Some collision shape in the URDF uses cylinder as its geometry and it is currently not supported. The workaround is to replace such occurrences with a mesh geometry that contains the triangle mesh for a cylinder.

**Reply by Charlie0257** [reply]
Thanks for your reply! :)

I have replaced the cylinder with capsule, but its visualization still blows up, which can work well in IsaacGym.

Could you offer me some possible solutions?

**Comment by Charlie0257** [comment, marked as answer]
In fact, the explosion display of ur10e is due to the incompatibility between the corresponding mesh file and the Sapien simulator. Therefore, you can try other mesh files for ur10e on GitHub.

**Reply by fbxiang** [reply]
In the next major release (SAPIEN 3, currently in alpha). URDF cylinder tag will be supported (by creating a triangle mesh that looks like a cylinder under the hood). 
However, do note that some URDFs from Isaac (especially ones containing DAE files) will still not be compatible since Isaac seems to load DAE files differently from other libraries.
